src/auth/src/credentials/user_credential.rs

```diff
@@ -134,7 +137,15 @@ where
         let mut value = HeaderValue::from_str(&format!("{} {}", token.token_type, token.token))
             .map_err(|e| CredentialError::new(false, e.into()))?;
         value.set_sensitive(true);
-        Ok(vec![(AUTHORIZATION, value)])
+        let mut headers = vec![(AUTHORIZATION, value)];
+        if let Some(project) = &self.quota_project_id {
+            headers.push((
+                HeaderName::from_static(QUOTA_PROJECT_KEY),
+                HeaderValue::from_str(project)
+                    .map_err(|e| CredentialError::new(false, e.into()))?,
```

Comment:
Maybe as a future improvement: you could do this once when the credentials are created. Or cache the results on the first attempt and keep them from that point.

Reply:
I considered that, but I did not think there were any savings. If we cache the results, we still need to copy them.

The only thing I see is that we could reserve capacity in the vector, knowing we will add extra headers. That seemed negligible to me. :shrug:

Maybe I am missing something?

Reply:
(Aside) note to self: I should clean up the `map_err(...)` after #662

Reply:
I think you are right.